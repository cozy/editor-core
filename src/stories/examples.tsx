import React from 'react';

import { boolean, object, radios, withKnobs } from '@storybook/addon-knobs';
import { IntlProvider } from 'react-intl-next';

import Editor, { EditorProps } from '../editor';
import EditorContext from '../ui/EditorContext';

function WrapperEditorComponent(props: EditorProps) {
  const ChildKey = JSON.stringify(props);
  return (
    <EditorContext>
      <IntlProvider locale="en">
        <Editor key={ChildKey} {...props} />
      </IntlProvider>
    </EditorContext>
  );
}

export default {
  title: 'Editor',
  component: Editor,
  decorators: [withKnobs],
};

const groupId = 'GROUP-ID1';
const Template = (args: any) => (
  <WrapperEditorComponent
    allowUndoRedoButtons={boolean('Undo/Redo Buttons', false, groupId)}
    allowPanel={boolean('Panel', false, groupId)}
    appearance={radios(
      'Appearance',
      {
        Comment: 'comment',
        'Full Page': 'full-page',
      },
      'full-page',
      groupId,
    )}
    allowTables={object('Table Options', {}, groupId)}
    {...args}
  />
);

export const EditorExample = Template.bind({});
