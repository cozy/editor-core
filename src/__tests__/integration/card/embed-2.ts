import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { getDocFromElement, editable } from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { ConfluenceCardProvider } from '../../../../examples/5-full-page-with-confluence-smart-cards';
import * as embedCardAdf from './_fixtures_/embed-card.adf.json';
import { waitForEmbedCardSelection } from '@atlaskit/media-integration-test-helpers';

type ClientType = Parameters<typeof goToEditorTestingExample>[0];

BrowserTestCase(
  'card: changing the link URL of an embed link should convert it to a "dumb" link',
  { skip: ['safari', 'edge'] },
  async (client: ClientType, testName: string) => {
    const page = await goToEditorTestingExample(client);

    const cardProviderPromise = Promise.resolve(
      new ConfluenceCardProvider('prod'),
    );

    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
      defaultValue: JSON.stringify(embedCardAdf),
      UNSAFE_cards: {
        provider: cardProviderPromise,
        allowBlockCards: true,
        allowEmbeds: true,
      },
    });

    await waitForEmbedCardSelection(page);
    await page.click('button[aria-label="Edit link"]');
    // Clear the Link Label field before typing
    await page.clear('[data-testid="link-url"]');
    // Change the 'link address' field to another link and press enter
    await page.type('[data-testid="link-url"]', [
      'https://www.atlassian-new.com',
      'Return',
    ]);

    expect(
      await page.$eval(editable, getDocFromElement),
    ).toMatchCustomDocSnapshot(testName);
  },
);
