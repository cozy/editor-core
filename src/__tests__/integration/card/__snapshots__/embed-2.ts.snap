// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card: changing the link URL of an embed link to an unsupported url should convert it to a "dumb" link 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "data": Object {
              "@context": "https://www.w3.org/ns/activitystreams",
              "@type": "Document",
              "name": "https://product-fabric.atlassian.net/wiki/spaces/E/overview",
              "url": "https://product-fabric.atlassian.net/wiki/spaces/E/overview",
            },
            "url": null,
          },
          "type": "inlineCard",
        },
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
